engine/Shopware/Components/Translation.php

```diff
@@ -391,6 +397,327 @@ public function write($language, $type, $key = 1, $data = null, $merge = false)
+     *          ...
+     *          [ 'source' => 'desc', 'destination' => 'desc', 'default' => 'name' ],
+     *          ...
+     *      ];
+     *      In case no translation exists, the translated key will be filled with a default value
+     *      of the object. Therefore the translated key will always contain a value, either a translation
+     *      or the fallback value.
+     * Please notice, if a key of an object already has content and a translation is mapped onto that key,
+     * the original content will be overridden.
+     *
+     * Function call:
+     * $translatedPayments = $this->translateObjects(
+     *      // all objects to be translated
+     *      [ $payment, $payment1, ... ],
+     *
+     *      // the objecttype of the translations in s_core_translations
+     *      'config_payment',
+     *
+     *      // The context which the current calling context (for language selection)
+     *      Shopware_Components_Translation::CONTEXT_BACKEND,
+     *
+     *      // Array to map the translation onto the objects
+     *      $translationMapping
+     * );
+     *
+     * Result:
+     * $translatedPayments = [
+     *      [
+     *          // Key which are not mapped to will NOT be changed
+     *          'id' => 5,
+     *          'name' => 'debit',
+     *
+     *          'description' => 'Debit',
+     *          'additionalDescription' => 'Insert text',
+     *
+     *           // Additional values
+     *          ...
+     *      ]
+     * ];
+     *
+     * @param array  $objects               Objects to be translated
+     * @param string $type                  Objecttype of s_core_translations
+     * @param int    $context               Calling context CONTEXT_BACKEND, CONTEXT_FRONTEND
+     * @param array  $translationMappingArg How the translation should be mapped onto the objects
+     * @param int    $language              Language id
+     * @param int    $fallback              Id of fallback language
+     *
+     * @return array Translated objects
+     */
+    public function translateObjects($objects, $type, $context, $translationMappingArg, $language = null, $fallback = null)
+    {
+        // Check if the languages are specified and query them if not
+        if ($language === null && $fallback === null) {
+            // Query the correct languages
+            $languageArr = $this->getLanguageIdsByContext($context);
+            $language = $languageArr['main'];
+            $fallback = $languageArr['fallback'];
+        }
+
+        // Load translations from db
+        $translations = $this->readBatchWithFallback($language, $fallback, $type);
+
+        // Create correct mapping array from parameter
+        $mappingLambda = function ($value) {
+            if (is_array($value)) {
+                return $value;
+            }
+
+            return [
+                    'source' => $value,
+                    'destination' => $value,
+                ];
```

Comment:
Indentation...

Reply:
As discussed personally, the indentation was created by shopware's git hook.